app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -149,4 +149,22 @@ def relation_values(association, target)
 
     top_level && (target == true || target.nil? || parent == target) ? :use_association : []
   end
+
+  def get_cluster(ems, cluster_hash, rp_hash, dc_hash)
+    cluster = EmsCluster.find_by(:ems_ref => cluster_hash[:ems_ref], :ems_id => ems.id)
+    if cluster.nil?
+      rp = ems.resource_pools.create!(rp_hash)
+
+      cluster = ems.clusters.create!(cluster_hash)
+
+      cluster.add_resource_pool(rp)
+      cluster.save!
+
+      dc = Datacenter.find_by(:name => dc_hash[:name], :ems_id => ems.id)
```

Comment:
The name guarantees uniqueness of the DC in rhev. Potentially we could use ems_ref here if we wanted to.